modules/core/misc/java/test/MatOfByteTest.java

```diff
@@ -0,0 +1,94 @@
+package org.opencv.test.core;
+
+import java.util.Arrays;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.Path;
+import java.io.File;
+import java.io.IOException;
+
+import org.opencv.core.Core;
+import org.opencv.core.CvException;
+import org.opencv.core.CvType;
+import org.opencv.core.Mat;
+import org.opencv.core.MatOfByte;
+import org.opencv.core.MatOfDouble;
+import org.opencv.test.OpenCVTestCase;
+import org.opencv.test.OpenCVTestRunner;
+import org.opencv.imgcodecs.Imgcodecs;
+
+public class MatOfByteTest extends OpenCVTestCase {
+    public static String LENA_W_10BYTEMARGIN_PATH = "";
+    public static String CHESS_W_10BYTEMARGIN_PATH = "";
+    Mat rgbLena2;
+    Mat grayChess2;
+    Mat rgbLenawMargin;
+    Mat grayChesswMargin;
+    
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        LENA_W_10BYTEMARGIN_PATH = OpenCVTestRunner.LENA_PATH.replace("lena.png", "lena-10byte_margin.png");
+        CHESS_W_10BYTEMARGIN_PATH = OpenCVTestRunner.CHESS_PATH.replace("chessboard.jpg", "chessboard-10byte_margin.jpg");
```

Comment:
These new files are malformed.
Please simplify test without imdecode() calls. We should test this functionality, not imdecode().

Something like this:
```
byte[] inputBytes = new byte[5] { 1,2,3,4,5 };
May m = new Mat(inputBytes);
May m0 = new Mat(0, 5, inputBytes);
assertArrayEquals(m, m0, EPS);
```

```
May m1 = new Mat(2, 2, inputBytes);
assertEquals(3, m1.get(0));
```

Reply:
I see, tests need to be as independent as possible. I will fix this.

Reply:
The files are malformed for a reason. My initial goal was to extract images embedded in RAW files coming from my DSLR but giving an offset and length of the JPG file. I added random bytes before and after the files to see if I can extract the png or jpg encoded files out. But IMO your example should be sufficient to test this. So, I will simplify it.